### DISCUSSION TITLE
Specifying Default Dropdown Selection Value

### DISCUSSION THREAD

**Discussion by rev9tech** [Power-Components/livewire-powergrid#400] [General]
Hello:
Is there a way to specify a default dropdown selection when the grid is rendered?  For example: I would like to initially show a powergrid for items on a list that are "Active." There is a drop-down that contains the values "Active/Inactive."  It would be great to initially display "Active" parts by default but allow the user to selective inactive if they choose.  Right now, it shows "All" parts by default.  Thanks in advance for any guidance you can provide.

**Comment by pieterschreuder** [comment]
Any thoughts about this idea? I would use this feature as well.

**Comment by luanfreitasdev** [comment]
Wouldn't it be in the query query using ->where('active', true) ?

**Reply by ZulfikarHD** [reply]
hi, i'm using Filter:datepicker, i also want the filter to have default value of this month range, i'm trying using query ->whereBetween(date,[startMonth,endMonth]) it works, but if you change the daterange outside the initial query, eg.last month range, it will not shown the data at all it only works between the initial query daterange.

**Comment by pieterschreuder** [comment]
That will show all active items on initial page load, but:

- Doesn't change the default filter value
- If you choose 'all' it wouldn't show active items and inactive items (because of the query where condition).

**Reply by luanfreitasdev** [reply]
Good observation. Any suggestions to fix this problem?

**Reply by pieterschreuder** [reply]
It would suggest to add a function `setDefaultValue(..)` to `PowerComponents\LivewirePowerGrid\Column`. 

Defining a column could look like this:

`Column::add()->makeBooleanFilter('active', 'Active', 'Inactive')->defaultFilterValue(...)`.

**Reply by rev9tech** [reply]
@pieterschreuder Totally agree with this recommended approach!

**Reply by rev9tech** [reply]
Just checking in on this.  Would it be hard to add this feature?

**Reply by ashek1412** [reply]
Hi, I am looking for this feature also. any update

**Reply by dwainscheeren** [reply]
Still not possible?

**Comment by fabryss** [comment]
I have solved using the default value in the boot function like this:

`public function filters(): array
    {
        return [
            Filter::boolean('is_active')->label('Yes', 'No')
        ];
    }
`

for default "no" boolean i set:

`public function boot(): void
    {
        $this->filters['boolean']['is_active'] = false;
    }`

I hope it can help :)

**Reply by ZulfikarHD** [reply]
i cant make it work for datepicker, even tough i can made the default datepicker value, but the base table dont follow the datepicker filter for the initial boot,

**Reply by daniieljc** [reply]
How could you set a default value in the datepicker? I've been researching for days but I can't find the solution.
